File: app/Models/Pembayaran.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Pembayaran extends Model
{
    use HasFactory;

    protected $table = 'pembayaran';

    protected $fillable = [
        'tagihan_id',
        'tanggal_bayar',
        'jumlah_bayar',
        'metode_bayar',
        'penyedia_layanan',
        'kode_pembayaran',
        'status_pembayaran',
        'referensi_gateway',
        'bukti_pembayaran',
        'petugas_id',
        'expired_at',
    ];

    protected $casts = [
        'tanggal_bayar' => 'datetime',
        'expired_at' => 'datetime',
    ];

    public function tagihan()
    {
        return $this->belongsTo(Tagihan::class);
    }

    public function petugas()
    {
        return $this->belongsTo(User::class, 'petugas_id');
    }
}

File: mobile-flutter/lib/models/pembayaran.dart
import 'tagihan.dart';

class Pembayaran {
  final int id;
  final int tagihanId;
  final String? tanggalBayar;
  final double jumlahBayar;
  final String metodeBayar;
  final String? penyediaLayanan;
  final String kodePembayaran;
  final String statusPembayaran;
  final String? referensiGateway;
  final String? idTransaksi;
  final DateTime? expiredAt;
  final Tagihan? tagihan;

  Pembayaran({
    required this.id,
    required this.tagihanId,
    this.tanggalBayar,
    required this.jumlahBayar,
    required this.metodeBayar,
    this.penyediaLayanan,
    required this.kodePembayaran,
    required this.statusPembayaran,
    this.referensiGateway,
    this.idTransaksi,
    this.expiredAt,
    this.tagihan,
  });

  factory Pembayaran.fromJson(Map<String, dynamic> json) {
    return Pembayaran(
      id: json['id'],
      tagihanId: json['tagihan_id'],
      tanggalBayar: json['tanggal_bayar'],
      jumlahBayar: double.tryParse(json['jumlah_bayar'].toString()) ?? 0,
      metodeBayar: json['metode_bayar'] ?? '',
      penyediaLayanan: json['penyedia_layanan'],
      kodePembayaran: json['kode_pembayaran'] ?? '',
      statusPembayaran: json['status_pembayaran'] ?? 'Pending',
      referensiGateway: json['referensi_gateway'],
      idTransaksi: json['id_transaksi']?.toString(),
      expiredAt: json['expired_at'] != null 
          ? DateTime.tryParse(json['expired_at']) 
          : null,
      tagihan: json['tagihan'] != null
          ? Tagihan.fromJson(json['tagihan'])
          : null,
    );
  }
}
